<?xml version="1.0"?>
<robot name="Unicycle">
  <link name="base_link"/>
  <link name="carbody">
			<visual>
				<origin xyz="0 0 0" rpy="0 0 0" />
				<geometry>
					<box size="0.38 0.5 0.2"/>
				</geometry>
		 </visual>
     <!-- Leaving inertial stuff as is right now -->
			<inertial>
				<mass value="10"/>
				<inertia ixx="0.4" ixy="0.0" ixz="0.0" iyy="0.4" iyz="0.0" izz="0.2"/>
			</inertial>
  </link>
<link name="backwheel1">
    <visual>
      <geometry>
        <cylinder length="0.06" radius="0.07"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 0.8 1"/>
      </material>
    </visual>
  </link>
  <link name="backwheel2">
    <visual>
      <geometry>
        <cylinder length="0.06" radius="0.07"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 0.8 1"/>
      </material>
    </visual>
  </link>
  <link name="frontwheel1">
    <visual>
      <geometry>
        <cylinder length="0.06" radius="0.07"/>
      </geometry>
      <material name="red">
        <color rgba="0.8 0 0 1"/>
      </material>
    </visual>
  </link>
  <link name="frontwheel2">
    <visual>
      <geometry>
        <cylinder length="0.06" radius="0.07"/>
      </geometry>
      <material name="red">
        <color rgba="0.8 0 0 1"/>
      </material>
    </visual>
  </link>
  <!-- Joints -->
	<joint name="basetocarbody" type="fixed">
    <parent link="base_link"/>
    <child link="carbody"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
	</joint>
  <joint name="base_to_backwheel1" type="continuous">
    <parent link="carbody"/>
    <child link="backwheel1"/>
    <origin rpy="0 -1.5707 0" xyz="-0.22 -0.25 -0.08"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="base_to_frontwheel1" type="continuous">
    <parent link="carbody"/>
    <child link="frontwheel1"/>
    <origin rpy="0 -1.5707 0" xyz="-0.22 0.25 -0.08"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="base_to_frontwheel2" type="continuous">
    <parent link="carbody"/>
    <child link="frontwheel2"/>
    <origin rpy="0 -1.5707 0" xyz="0.22 0.25 -0.08"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="backwheel1_to_backwheel2" type="fixed">
    <parent link="backwheel1"/>
    <child link="backwheel2"/>
    <origin rpy="0 0 0" xyz="0 0 -0.450"/>
  </joint>
</robot>
